sns/sns.go

```diff
@@ -0,0 +1,137 @@
+// Attribution: robbiet480's sns.go repo (https://github.com/robbiet480/go.sns) was used as the starting point for this file under the MIT License.
+
+// Package sns provides AWS SNS related functionality.
+package sns
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+	"time"
+
+	"github.com/go-playground/errors/v5"
+)
+
+var hostPattern = regexp.MustCompile(`^sns\.[a-zA-Z0-9\-]{3,}\.amazonaws\.com(\.cn)?$`)
+
+type Payload struct {
+	Message          string `json:"Message"`
+	MessageID        string `json:"MessageId"`
+	Signature        string `json:"Signature"`
+	SignatureVersion string `json:"SignatureVersion"`
+	SigningCertURL   string `json:"SigningCertURL"`
+	SubscribeURL     string `json:"SubscribeURL"`
+	Subject          string `json:"Subject"`
+	Timestamp        string `json:"Timestamp"`
+	Token            string `json:"Token"`
+	TopicArn         string `json:"TopicArn"`
+	Type             string `json:"Type"`
+	UnsubscribeURL   string `json:"UnsubscribeURL"`
+}
+
+type Client struct{}
+
+func New() *Client {
+	return &Client{}
+}
+
+// VerifyAuthenticity verifies that the body of the request is authentic (i.e., that it was sent by AWS SNS) and returns the payload.
+func (c *Client) VerifyAuthenticity(ctx context.Context, reqBody []byte) (*Payload, error) {
```

Comment:
Should `reqBody` be a reader here? Not sure how we are getting the data out of the SNS message?